### ISSUE TITLE
Unclear error handling in StreamOnce

### ISSUE THREAD

**Issue by tailhook** [Marwes/combine#131]
StreamOnce has an associated type but requires `uncons` to be generic over the error type:
```
type Error: ParseError<Self::Item, Self::Range, Self::Position>
fn uncons<E>(&mut self) -> Result<Self::Item, E> where
    E: StreamError<Self::Item, Self::Range>, 
```
I don't understand why so? It's probably a documentation issue or might be a bug?

I'm using combine=3.0.0-alpha.5

**Comment by Marwes** [comment]
Thanks for picking up on this. I reverted this so that the method is non-generic again. It used to be necessary to avoid losing error information but I realized now that it is only necessary in a select few cases.

See #132

**Comment by Marwes** [comment]
Published 3.0.0-beta.1 with #132
